src/librustc_typeck/diagnostics.rs

```diff
@@ -293,6 +373,29 @@ return, for example with a `loop` that never breaks or a call to another
 diverging function (such as `panic!()`).
 "##,
 
+E0178: r##"
+In types, the `+` type operator has low precedence, so it is often necessary
+to use parentheses.
+
+For example:
+
+```
+trait Foo {}
+
+struct Bar<'a> {
+    w: &'a Foo + Copy,   // error, use &'a (Foo + Copy)
+    x: &'a Foo + 'a,     // error, use &'a (Foo + 'a)
+    y: &'a mut Foo + 'a, // error, use &'a mut (Foo + 'a)
+    z: fn() -> Foo + 'a, // error, use fn() -> (Foo + 'a)
+}
+```
+
+More details can be found in [RFC 438].
+
+[RFC 438]: https://github.com/rust-lang/rfcs/blob/master/text/0438-precedence-\
+of-plus.md
```

Comment:
Nope! I don't know what I was thinking (or why I didn't test it). It's a bit unfortunate though, I'm not sure that it's possible to include long URLs in this format.

I've changed it to a shorter URL.